api-matlab/wrapper_physiobatch.m

```diff
@@ -0,0 +1,111 @@
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {''}; 
+     elseif strcmp(data_vendor, 'Siemens_Tics')
+        [cardiac, ~] = spm_select(1,'.*.log','Specify cardiac file (..._PULS.log)',{}, default_path);
+        % cardiac = input('Specify file path to the cardiac file (in template ending with _PULS.log)', 's');
+        [respiration, ~] = spm_select(1,'.*.log','Specify respiration file (..._RESP.log)',{}, default_path);
+        % respiration = input('Specify file path to the respiration file (in template ending with _RESP.log)', 's');
+        [timing, ~] = spm_select(1,'.*.log','Specify scan timing file (..._Info.log)',{}, default_path);
+        % timing = input('Specify file path to the scan timing file (in template ending with _Info.log)', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {cardiac};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {respiration};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {timing};
+    else
+        ME = MException('error:wronginput', 'Wrong vendor specified. Please specify one of BIDS, Philips, Biopac_Txt, GE, Siemens, Siemens_HCP, Siemens_Tics');
+        throw(ME);
+    end
+    
+    % read scan parameters
+    input_params = inputdlg({'Number of scans:','Number of slices:','Number of dummies:', 'TR:', 'Onset slice:'}, 'Parameters for physio');
+    
+    % scan parameters
+    matlabbatch{1}.spm.tools.physio.log_files.sampling_interval = [];
+    matlabbatch{1}.spm.tools.physio.log_files.relative_start_acquisition = 0;
+    matlabbatch{1}.spm.tools.physio.log_files.align_scan = 'last'; % sometimes first, sometimes last
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nslices = str2double(input_params{2});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.NslicesPerBeat = [];
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.TR = str2double(input_params{4});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Ndummies = str2double(input_params{3});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nscans = str2double(input_params{1});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.onset_slice = str2double(input_params{5});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.time_slice_to_slice = [];
+
+    % template
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nprep = [];
+    matlabbatch{1}.spm.tools.physio.scan_timing.sync.nominal = struct([]);
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.modality = 'ECG';
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.initial_cpulse_select.auto_matched.min = 0.4;
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.initial_cpulse_select.auto_matched.file = 'initial_cpulse_kRpeakfile.mat';
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.posthoc_cpulse_select.off = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.output_multiple_regressors = 'multiple_regressors.txt';
```

Comment:
I think we should name this something like "retroicor_regressors.txt" to make it clearer what is contained in the file, since "multiple_regressors.txt" is also standard in SPM itself.

Reply:
Sure, will do